WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRowViewModel.swift

```diff
@@ -63,6 +63,92 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
+
+    /// Description of the subscription billing details for a Subscription-type Product
+    /// eg: "$60.00 / 2 months"
+    ///
+    var subscriptionBillingDetailsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        guard let subscriptionPrice = productSubscriptionDetails?.price,
+              let subscriptionInterval = productSubscriptionDetails?.periodInterval,
+              let subscriptionPeriod = productSubscriptionDetails?.period,
+              let formattedPrice = currencyFormatter.formatAmount(subscriptionPrice, with: currency) else {
+            return ""
+        }
+
+        let subscriptionFrequency = {
+            switch subscriptionInterval {
+            case "1":
+                return subscriptionPeriod.descriptionSingular
+            default:
+                return subscriptionPeriod.descriptionPlural
+            }
+        }
+        return String.localizedStringWithFormat(Localization.formattedProductSubscriptionBilling,
+                                                formattedPrice, subscriptionInterval, subscriptionFrequency())
+    }
+
+    /// Description of the subscription conditions for a Subscription-type Product
+    /// These are separate from each other, a subscription could have any, all, or none
+    /// eg: "$25.00 signup · 1 month free"
+    ///
+    var subscriptionConditionsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        // Signup fees
+        guard let signUpFee = productSubscriptionDetails?.signUpFee else {
+            return ""
+        }
+        var formattedSignUpFee: String = ""
+        if signUpFee.isEmpty {
+            formattedSignUpFee = ""
+        } else {
+            formattedSignUpFee = currencyFormatter.formatAmount(signUpFee, with: currency) ?? ""
+        }
+
+        // Trial periods
+        guard let trialLength = productSubscriptionDetails?.trialLength,
+              let trialPeriod = productSubscriptionDetails?.trialPeriod else {
+            return ""
+        }
+
+        let formattedTrialDetails = {
+            switch trialLength {
+            case "":
+                // The API allows an empty value for trial length, with a non-nil trial period.
+                // eg: every -empty- days
+                return ""
+            case "0":
+                // The API allows to input a 0-length trial length, with a non-nil trial period.
+                // eg: every zero days
+                return "0"
+            case "1":
+                return trialPeriod.descriptionSingular
+            default:
+                return trialPeriod.descriptionPlural
+            }
+        }
+
+        let hasNoSignUpFees = formattedSignUpFee.isEmpty || formattedSignUpFee == "0"
+        let hasNoFreeTrial = formattedTrialDetails().isEmpty || formattedTrialDetails() == "0"
```

Comment:
Nit: similar comment as above about using computed constant instead of closure:

```suggestion
        }()

        let hasNoSignUpFees = formattedSignUpFee.isEmpty || formattedSignUpFee == "0"
        let hasNoFreeTrial = formattedTrialDetails.isEmpty || formattedTrialDetails == "0"
```

Reply:
Yep! Updated along the other instance on e92b901266ba7dadfb93b4a88a31da7a00641486